pkg/cloud/services/networking/securitygroups.go

```diff
@@ -56,14 +56,26 @@ func (s *Service) ReconcileSecurityGroups(clusterName string, openStackCluster *
 		s.logger.V(4).Info("No need to reconcile security groups", "cluster", clusterName)
 		return nil
 	}
+
+	secControlPlaneGroupName := fmt.Sprintf("%s-cluster-%s-secgroup-%s", secGroupPrefix, clusterName, controlPlaneSuffix)
+	var err error
+	err = s.createSecurityGroupIfNotExists(secControlPlaneGroupName)
+	if err != nil {
+		return nil
+	}
+	secWorkerGroupName := fmt.Sprintf("%s-cluster-%s-secgroup-%s", secGroupPrefix, clusterName, workerSuffix)
+	err = s.createSecurityGroupIfNotExists(secWorkerGroupName)
+	if err != nil {
+		return nil
+	}
+
 	desiredSecGroups := map[string]infrav1.SecurityGroup{
-		"controlplane": generateControlPlaneGroup(clusterName),
-		"global":       generateGlobalGroup(clusterName),
+		"controlplane": s.generateControlPlaneGroup(clusterName),
+		"worker":       s.generateWorkerGroup(clusterName),
 	}
 	observedSecGroups := make(map[string]*infrav1.SecurityGroup)
 
 	for k, desiredSecGroup := range desiredSecGroups {
-		s.logger.Info("Reconciling security group", "name", desiredSecGroup.Name)
```

Comment:
@hidekazuna Just curious. Won't it be better to keep this? Just to log which security group is being reconciled.

Reply:
@hidekazuna Got it. Thanks